packages/backend/src/app/flows/step-run/step-run-service.ts

```diff
@@ -144,34 +145,44 @@ const executeBranch = async ({ step, flowVersion, projectId }: ExecuteParams<Bra
         })
     }
 
-    delete branchStep.nextAction
-    delete branchStep.onFailureAction
-    delete branchStep.onSuccessAction
-
     const testTrigger: EmptyTrigger = {
-        name: 'test_branch_step',
+        name: 'test_trigger',
         valid: true,
         displayName: 'test branch step',
-        nextAction: branchStep,
+        nextAction: {
+            ...branchStep,
+            nextAction: undefined,
+            onSuccessAction: undefined,
+            onFailureAction: undefined,
+        },
         type: TriggerType.EMPTY,
         settings: {},
     }
 
-    flowVersion.trigger = testTrigger
+    const testFlowVersion: FlowVersion = {
```

Comment:
Isn't that the same issue, how it would extract the context? if it doesn't have the flow?

Reply:
Edit: why we are sending two flows? why not the origianl one only and use flowUtils to find the correct step?

Reply:
The second flow is an empty one containing the test branch only, this is a generic design that can handle multiple cases like: testing branches, testing code pieces, testing piece actions, and so on.

Other parts of the test step service will be migrated to this design after proving its correctness on the branch step. This will benefit the system in, mainly, two points:
- Reduce the number of operations executed by the engine
- Reduce duplication across different engine operations